### PR TITLE
JBPM-7632 - Enhance process definition data including timers and nodes

### PR DESCRIPTION
**Pull request by cristianonicolai**

### CODE REVIEW

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/xml/ServicesProcessDataEventListener.java [hunk 1]
```diff
@@ -126,7 +129,21 @@ public void onNodeAdded(Node node) {
             } else {
                 processDescriptor.getReusableSubProcesses().add(processId);
             }
+        } else if (node instanceof TimerNode) {
```
Comment (mswiderski) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/xml/ServicesProcessDataEventListener.java]: what about timer start events? do we want to cover them? Mainly in context of event subprocess...

Comment (cristianonicolai) [reply jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/xml/ServicesProcessDataEventListener.java]: @mswiderski makes sense to me, I included the _StartNode_ as well, thanks.

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/xml/ServicesProcessDataEventListener.java [hunk 2]
```diff
@@ -126,7 +130,24 @@ public void onNodeAdded(Node node) {
...
+            TimerNode timerNode = (TimerNode) node;
+            processDescriptor.getTimers().add(new TimerDesc(timerNode.getTimer().getId(), timerNode.getId(), (String) timerNode.getMetaData().get("UniqueId"), timerNode.getName()));
+        } else if (node instanceof StartNode) {
+            StartNode startNode = (StartNode) node;
+            processDescriptor.getTimers().add(new TimerDesc(startNode.getTimer().getId(), startNode.getId(), (String) startNode.getMetaData().get("UniqueId"), startNode.getName()));
```
Comment (nmirasch) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/xml/ServicesProcessDataEventListener.java]: Do we have to check if starNode.getTimer() could be null here?

**Comment by mswiderski** [comment]
Jenkins retest this

**Comment by tsurdilo** [comment]
Jenkins retest please

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/xml/ServicesProcessDataEventListener.java (hunk 2 above)
Comment (cristianonicolai) [reply jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/xml/ServicesProcessDataEventListener.java]: @nmirasch makes sense, thanks, fixed now.

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Pull request merged**